<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Gepoliano Chaves" />


<title>Download and align Fastq files and extract Read Counts</title>

<script src="site_libs/header-attrs-2.28/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ReComBio</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="about.html">
    <span class="fa fa-info"></span>
     
    About
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Code Notebooks
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Part 1: The R Syntax</li>
    <li>
      <a href="notebooks/Introduction-to-R-and-Terminal.html">1.1. Introduction to R</a>
    </li>
    <li>
      <a href="notebooks/01-neuroblastoma-r2-GSE62564.html">1.2. Data Download and Processing</a>
    </li>
    <li>
      <a href="notebooks/02-neuroblastoma-r2-plots-GSE62564.html">1.3. Data Visualization</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Part 2: Machine Learning</li>
    <li>
      <a href="align-extract-counts.html">2.1. Align and Extract Counts</a>
    </li>
    <li>
      <a href="transcriptomics-analysis.html">2.2. Transcriptome Analysis</a>
    </li>
    <li>
      <a href="notebooks/17-shiny-tumor-classification-bbc2024.html">2.3. Introduction to Machine Learning</a>
    </li>
    <li>
      <a href="notebooks/17-shiny-tumor-classification-bbc2024.html">2.4. Classification using Logistic Regression</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Part 3: DNA Sequencing and the Command Line</li>
    <li>
      <a href="https://www.youtube.com/watch?v=lU8ztLZRrvI">3.1. DNA Sequencing Technologies</a>
    </li>
    <li>
      <a href="download_sequences.html">3.2A. Download mtDNA fasta sequences</a>
    </li>
    <li>
      <a href="notebooks/07-Phylogenetic-Analyisis_mt_BR.html">3.2B. Phylogenetic Analysis</a>
    </li>
    <li>
      <a href="notebooks/haplogroups.html">3.2C. Results: mtDNA haplogroups</a>
    </li>
    <li>
      <a href="https://haplogrep.readthedocs.io/en/latest/annotations/#clusters-and-population-frequencies">3.2D. Haplogroups Annotations</a>
    </li>
    <li>
      <a href="notebooks/Introduction-to-the-Terminal.html">3.3. Command Line or Terminal</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Part 4: Presentations and Final Considerations</li>
    <li>
      <a href="02-variant_call_pipeline.html">4.1. Variant Call Pipeline using Sequenced DNA</a>
    </li>
    <li>
      <a href="exercises/exercises#chapter-2-command-line-or-terminal">4.2. Access terminal in AWS image</a>
    </li>
    <li>
      <a href="notebooks/10-Survival-KM-and-Cox-Regression-Kocak-498.html">4.3. Final Presentations</a>
    </li>
  </ul>
</li>
<li>
  <a href="videos.html">
    <span class="fa fa-solid fa-video"></span>
     
    Videos
  </a>
</li>
<li>
  <a href="pdf/miniproject-slides.pdf">
    <span class="fa fa-address-book-o"></span>
     
    Mini Project
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-address-book-o"></span>
     
    Exercises
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Exercises</li>
    <li>
      <a href="exercises/exercises.html">Questions</a>
    </li>
    <li>
      <a href="assignment-dates">Assignments by Date</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">Solutions</li>
    <li>
      <a href="exercises/solutions.pdf">Preliminary Solutions</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-address-book-o"></span>
     
    Final Presentations
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Final Projects</li>
    <li>
      <a href="final-presentations.html">Final-Presentations</a>
    </li>
    <li>
      <a href="notebooks/neuroblastoma-survival.html">Project 2: Survival Analysis of Neuroblastoma</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Download and align Fastq files and extract
Read Counts</h1>
<h4 class="author">Gepoliano Chaves</h4>
<h4 class="date">October 15th, 2024</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#references" id="toc-references">References</a></li>
<li><a href="#transfer-files" id="toc-transfer-files">Transfer
files</a></li>
<li><a href="#look-at-geo-metadata-file"
id="toc-look-at-geo-metadata-file">Look at GEO Metadata File</a></li>
<li><a href="#download-fastq-and-align"
id="toc-download-fastq-and-align">Download fastq and align</a></li>
<li><a href="#extract-counts-with-featurecounts"
id="toc-extract-counts-with-featurecounts">Extract counts with
featureCounts</a></li>
</ul>
</div>

<p><br />
</p>
<div id="references" class="section level2">
<h2>References</h2>
<p><br />
</p>
<p>Google search: SRA download subset number of reads</p>
<p>In the link below, I learned the syntax to download a subset of fastq
reads rather than an entire fastq file: <a
href="https://bioinformatics.ccr.cancer.gov/docs/b4b/Module1_Unix_Biowulf/Lesson6/"
class="uri">https://bioinformatics.ccr.cancer.gov/docs/b4b/Module1_Unix_Biowulf/Lesson6/</a></p>
<p><br />
</p>
</div>
<div id="transfer-files" class="section level2">
<h2>Transfer files</h2>
<p><br />
</p>
<div id="transfer-an-alignment-pipeline-script" class="section level4">
<h4>Transfer an alignment pipeline script</h4>
<pre class="bash"><code>scp gchaves@hb.ucsc.edu:/hb/home/gchaves/scripts/Signaling_GSE68952_H3K4me3.sh ./scripts</code></pre>
<p><br />
</p>
</div>
<div id="transfer-the-human-reference-genome" class="section level4">
<h4>Transfer the human reference genome</h4>
<pre class="bash"><code>origin=/hb/home/gchaves/references/hg19/GRCh37.p13.genome.fa
destination=/home/bioinfo/data/references/hg19/GRCh37.p13.genome.fa

/hb/home/gchaves/programs/sshpass-1.06/sshpass -p &quot;bioinfo&quot; \
scp $origin bioinfo@200.128.7.33:$destination</code></pre>
<p><br />
</p>
</div>
<div id="transfer-featurecounts" class="section level4">
<h4>Transfer featureCounts</h4>
<pre class="bash"><code>scp -r gchaves@hb.ucsc.edu:/hb/home/gchaves/results/neuroblastoma/SRR899360_1.fastq /home/bioinfo/data/</code></pre>
<p><br />
</p>
</div>
<div id="transfer-fastq-file" class="section level4">
<h4>Transfer fastq file</h4>
<pre class="bash"><code>scp -r /hb/home/gchaves/results/neuroblastoma/SRR899360_1.fastq bioinfo@200.128.7.33:/home/bioinfo/data/neuroblastoma</code></pre>
<p><br />
</p>
</div>
<div id="transfer-sam-file" class="section level4">
<h4>Transfer sam file</h4>
<pre class="bash"><code>scp -r /hb/home/gchaves/results/neuroblastoma/SRR899360.sam bioinfo@200.128.7.33:/home/bioinfo/data/neuroblastoma</code></pre>
<p><br />
</p>
</div>
<div id="transfer-gtf-file" class="section level4">
<h4>Transfer GTF file</h4>
<pre class="bash"><code>scp -r /hb/home/gchaves/references/hg19_bme237/gtf/gencode.v19.annotation.gtf  bioinfo@200.128.7.33:/home/bioinfo/data/references/</code></pre>
<p><br />
</p>
</div>
<div id="build-the-index" class="section level4">
<h4>Build the index</h4>
<pre class="bash"><code>cd /home/bioinfo/data/references/hg19/
bwa index -a bwtsw GRCh37.p13.genome.fa</code></pre>
<p><br />
</p>
</div>
</div>
<div id="look-at-geo-metadata-file" class="section level2">
<h2>Look at GEO Metadata File</h2>
<p><br />
</p>
<p>In this part I stored the metadata information to document the SRA
file that I am downloading. Later I can match this file with the patient
and RNA-Seq sample that it was extracted from.</p>
<pre class="r"><code>library(maditr)
sra_run_table_df &lt;- read.table(&quot;./scripts/SraRunTable.txt&quot;, sep = &quot;,&quot;, header = T)
sra_run_table_selected &lt;- sra_run_table_df %&gt;% dplyr::select(Run, BioProject, BioSample, Experiment, 
                                                             flowcell, Sample.Name, BARCODE)</code></pre>
<p><br />
</p>
<p>Check flowcells and Barcodes</p>
<pre class="r"><code>library(expss)
# cross_cases(sra_run_table_selected, flowcell)
# cross_cases(sra_run_table_selected, BioProject)
# cross_cases(sra_run_table_selected, BARCODE)</code></pre>
<p><br />
</p>
</div>
<div id="download-fastq-and-align" class="section level2">
<h2>Download fastq and align</h2>
<p><br />
</p>
<p>This command loads the sratoolkit to the server environment:</p>
<pre class="bash"><code>module load sratoolkit/3.0.0</code></pre>
<p><br />
</p>
<p>This command downloads a limited number of reads using the SRA ID
number that in this case is SRR899360:</p>
<pre class="bash"><code>## Declare human reference genome
## REFERENCE=~/references/hg19/GRCh37.p13.genome.fa
REFERENCE=/home/bioinfo/data/references/hg19/GRCh37.p13.genome.fa

## Download fastq file
fastq-dump --split-files -X 1000 SRR899360 --split-files SRR899360 \
--outdir /hb/home/gchaves/results/neuroblastoma

## Specify fastq file
FastqR1=/hb/home/gchaves/results/neuroblastoma/SRR899360_1.fastq
    
## Align
module load bwa
#bwa mem -t 4 $REFERENCE $FastqR1 &gt; /hb/home/gchaves/results/neuroblastoma/SRR899360.sam
bwa mem -t 4 /home/bioinfo/data/references/hg19/GRCh37.p13.genome.fa /home/bioinfo/data/neuroblastoma/SRR899360_1.fastq &gt; /home/bioinfo/data/neuroblastoma/SRR899360.bam</code></pre>
<p><br />
</p>
</div>
<div id="extract-counts-with-featurecounts" class="section level2">
<h2>Extract counts with featureCounts</h2>
<p><br />
</p>
<p>Finally we export the number of counts to the counts file. According
to this reference, p is an option for paired-end fastq files:</p>
<ul>
<li><p><a
href="https://rnnh.github.io/bioinfo-notebook/docs/featureCounts.html"
class="uri">https://rnnh.github.io/bioinfo-notebook/docs/featureCounts.html</a></p></li>
<li><p>Bioinfo server</p></li>
</ul>
<pre class="bash"><code>/home/bioinfo/data/programs/subread-2.0.6-Linux-x86_64/bin/featureCounts -T 16 -a \
/home/bioinfo/data/references/gencode.v19.annotation.gtf -t gene -g \
gene_name -o /home/bioinfo/data/neuroblastoma/SRR899360.counts \
/home/bioinfo/data/neuroblastoma/SRR899360.sam</code></pre>
<p><br />
</p>
<p>Inspect the counts file:</p>
<pre class="bash"><code>cd /hb/home/gchaves/results/neuroblastoma/
head SRR899360.counts
wc -l SRR899360.counts</code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
